### PR TITLE
Issue 12

### PR DESCRIPTION
**Pull request by mgazz**
This PR is a follow up of #15 
- It updates the handlers processing the `AggregationSourceDiscovered` and `ResourceCreated` events. At the reception of the `AggregationSourceDiscovered` event the handler adds a UUID as `Context` to the Subscription. The Agent will have to add such `Context` to the following Events issued to the Sunfish server.
- It adds a Makefile simplifying the steps to build the `whl`

### CODE REVIEW

**Comment by christian-pinto** [comment]
@mgazz this needs to be rebased on the current master. Also, all commits must be signed. See the DCO check failing.

**Comment by mgazz** [comment]
@christian-pinto thank you for the review. I rebased the branch and added verified commits. I'll be available in case on any question.

**Review by cayton — APPROVED** [review]

**Review by cayton — APPROVED** [review]

File: sunfish/events/redfish_event_handler.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def check_subdirs(self, origin):
         
         return to_forward
     
-    def AggregationSourceDiscovered(self, event):
+    def AggregationSourceDiscovered(self, event,context):
```
Comment (dledford) [inline sunfish/events/redfish_event_handler.py:142]: Please don't mix whitespace styles (space after comma and no space after comma) in a single line.  I'm OK with either style, but we should be consistent.

File: sunfish/events/redfish_event_handler.py [hunk 2]
```diff
@@ -179,56 +179,35 @@ def AggregationSourceDiscovered(self, event):
...
+        add_aggregation_source_reference(object,aggregation_source)
 
-        self.create_object(id, response)
+        self.create_object(id, object)
 
-        RedfishEventHandler.bfsInspection(self, response, aggregation_source)
+        RedfishEventHandler.bfsInspection(self, object, aggregation_source)
 
-        self.patch_object(aggregation_source)
+        self.patch_object(id,aggregation_source)
```
Comment (dledford) [inline sunfish/events/redfish_event_handler.py:210]: Space after comma is the predominant style, please stick with that.

File: sunfish/events/redfish_event_handler.py [hunk 3]
```diff
@@ -335,14 +314,18 @@ def createInspectedObject(self,redfish_obj, aggregation_source):
...
-                        "@odata.id": aggregation_source["@odata.id"]
-                    }
-                }
-                if "Oem" not in redfish_obj:
-                    redfish_obj["Oem"] = {}
-                if "Sunfish_RM" not in redfish_obj["Oem"]:
-                    redfish_obj["Oem"]["Sunfish_RM"] = oem
-                self.create_object(file_path, redfish_obj)
\ No newline at end of file
+                add_aggregation_source_reference(redfish_obj,aggregation_source)
```
Comment (dledford) [inline sunfish/events/redfish_event_handler.py:317]: Space after comma is the predominant style, please stick with that.

File: sunfish/lib/core.py [hunk 4]
```diff
@@ -115,7 +115,7 @@ def replace_object(self, payload):
         # Call function from backend
         return self.storage_backend.replace(payload)
     
-    def patch_object(self, payload):
+    def patch_object(self, path,payload):
```
Comment (dledford) [inline sunfish/lib/core.py:118]: Space after comma is the predominant style, please stick with that.

File: sunfish/lib/core.py [hunk 5]
```diff
@@ -135,7 +135,7 @@ def patch_object(self, payload):
             return resp
                     
         # call function from backend
-        return self.storage_backend.patch(payload)
+        return self.storage_backend.patch(path,payload)
```
Comment (dledford) [inline sunfish/lib/core.py:138]: Space after comma is the predominant style, please stick with that.

File: sunfish/lib/core.py [hunk 6]
```diff
@@ -156,16 +156,22 @@ def delete_object(self, path):
...
     def handle_event(self, payload):
+        if "Context" in payload:
+            context = payload["Context"]
+        else:
+            context = ""
         for event in payload["Events"]:
             try:
                 handlerfunc = getattr(RedfishEventHandler, event['MessageId'].split(".")[-1])
-                handlerfunc(self, event)
+                handlerfunc(self, event,context)
```
Comment (dledford) [inline sunfish/lib/core.py:166]: Space after comma is the predominant style, please stick with that.

File: sunfish/lib/exceptions.py [hunk 7]
```diff
@@ -17,6 +17,23 @@ def __init__(self, resource_id):
...
+        Exception raised when the resource is not found.
+
+        Attributes:
+        attribute_name -- name of the attribute not found
+        message -- explanation of the error
++    """
+
+    def __init__(self, attribute_name):
+        self.attribute_name=attribute_name
+        message = "Attribute " + attribute_name+ " not found."
```
Comment (dledford) [inline sunfish/lib/exceptions.py:32]: Consistent whitespace please.  Put a space between attribute_name and +

File: sunfish/storage/backend_FS.py [hunk 8]
```diff
@@ -157,9 +157,11 @@ def replace(self, payload:json):
         except ResourceNotFound as e:
             raise ResourceNotFound(e.resource_id)
     
-    def patch(self, payload:json):
+    def patch(self, path: str, payload:json):
```
Comment (dledford) [inline sunfish/storage/backend_FS.py:160]: Either a space after the : or not, but not both

**Comment by mgazz** [comment]
@dledford thank you for the feedback. I agree with your request for changes. I will update as requested. 

Your review bring up something that we have yet to discuss: styling. We have yet to tackle this problem, we can adopt a linter that will help us styling the code automatically (example: pylint,flake8,etc..). (We can discuss this in another issue).

File: sunfish/lib/core.py (hunk 4 above)
Comment (mgazz) [reply sunfish/lib/core.py:118]: Fixed in pull request #15

File: sunfish/lib/core.py (hunk 5 above)
Comment (mgazz) [reply sunfish/lib/core.py:138]: Fixed in pull request #15

File: sunfish/lib/exceptions.py (hunk 7 above)
Comment (mgazz) [reply sunfish/lib/exceptions.py:32]: Fixed in pull request #15

File: sunfish/storage/backend_FS.py (hunk 8 above)
Comment (mgazz) [reply sunfish/storage/backend_FS.py:160]: Fixed in pull request #15

File: sunfish/events/redfish_event_handler.py [hunk 9]
```diff
@@ -179,56 +179,35 @@ def AggregationSourceDiscovered(self, event):
...
+        aggregation_source = self.get_object(
+            os.path.join(self.conf["redfish_root"], "AggregationService", "AggregationSources", context))
         hostname = aggregation_source["HostName"]
 
         response = requests.get(f"{hostname}/{id}")
         if response.status_code != 200:
             raise Exception("Cannot find ConnectionMethod")
-        response = response.json()
+        object = response.json()
+        add_aggregation_source_reference(object,aggregation_source)
```
[the remark this answers is not in the document]
Comment (mgazz) [reply sunfish/events/redfish_event_handler.py]: Fixed in fc3c6d61035d360216ee2d6448e425864598fec3

File: sunfish/events/redfish_event_handler.py (hunk 2 above)
Comment (mgazz) [reply sunfish/events/redfish_event_handler.py]: Fixed in fc3c6d61035d360216ee2d6448e425864598fec3

File: sunfish/lib/core.py (hunk 6 above)
Comment (mgazz) [reply sunfish/lib/core.py]: Fixed in 76ce2f5e1828ade356b95fda73d6ab262eba007f

**Review by dledford — APPROVED** [review]
Thanks for updating the style issues

**Pull request merged**